### ISSUE TITLE
Crash when sleeping or when time moves too fast

### ISSUE THREAD

**Issue by AbortRetryFail** [anegostudios/VintageStory-Issues#4166]
### Game Version

v1.20.0-pre.2

### Platform

Linux

### Modded

None

### SP/MP

Singleplayer

### Description

Occasionally, the game will crash with this message:
`System.IndexOutOfRangeException: Index was outside the bounds of the array.`

See log section for the whole message.

### How to reproduce

I don't know how. This one occurred after waking up from sleep.

### Logs

```Running on 64 bit Linux (Arch Linux) [Kernel 6.10.10.1] with 15965 MB RAM
Game Version: v1.20.0-pre.2 (Unstable)
9/26/2024 9:54:51AM: Critical error occurred
Loaded Mods: game@1.20.0-pre.2, creative@1.20.0-pre.2, survival@1.20.0-pre.2
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.API.MathTools.SimplexNoiseOctave.Evaluate(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexOctave.cs:line 414
   at Vintagestory.API.MathTools.SimplexNoise.Noise(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexNoise.cs:line 117
   at Vintagestory.GameContent.WeatherPattern.RegenCloudTileCache(Vec3i tileOffset) in VSEssentials\Systems\Weather\Model\Impl\WeatherPattern.cs:line 89
   at Vintagestory.GameContent.CloudRenderer.UpdateCloudTilesOffThread(Int32 changeSpeed) in VSEssentials\Systems\Weather\CloudRenderer.cs:line 390
   at Vintagestory.GameContent.CloudRenderer.<.ctor>b__31_2() in VSEssentials\Systems\Weather\CloudRenderer.cs:line 158
```

**Comment by AbortRetryFail** [comment]
I got this one again, this time while paddling a raft while hosting a LAN multiplayer game.
```
Running on 64 bit Linux (Arch Linux) [Kernel 6.10.10.1] with 15965 MB RAM
Game Version: v1.20.0-pre.2 (Unstable)
9/26/2024 8:19:18PM: Critical error occurred
Loaded Mods: game@1.20.0-pre.2, creative@1.20.0-pre.2, survival@1.20.0-pre.2
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.API.MathTools.SimplexNoiseOctave.Evaluate(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexOctave.cs:line 414
   at Vintagestory.API.MathTools.SimplexNoise.Noise(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexNoise.cs:line 117
   at Vintagestory.GameContent.WeatherPattern.RegenCloudTileCache(Vec3i tileOffset) in VSEssentials\Systems\Weather\Model\Impl\WeatherPattern.cs:line 89
   at Vintagestory.GameContent.WeatherDataReaderBase.ensureCloudTileCacheIsFresh(Vec3i tilePos) in VSEssentials\Systems\Weather\WeatherDataReader.cs:line 275
   at Vintagestory.GameContent.CloudRenderer.UpdateCloudTilesOffThread(Int32 changeSpeed) in VSEssentials\Systems\Weather\CloudRenderer.cs:line 390
   at Vintagestory.GameContent.CloudRenderer.<.ctor>b__31_2() in VSEssentials\Systems\Weather\CloudRenderer.cs:line 166
```

**Comment by AbortRetryFail** [comment]
Still happens on pre.5. This was after sleeping on a multiplayer game.
```
Running on 64 bit Linux (Arch Linux) [Kernel 6.10.10.1] with 15965 MB RAM
Game Version: v1.20.0-pre.5 (Unstable)
10/6/2024 8:58:18PM: Critical error occurred
Loaded Mods: game@1.20.0-pre.5, creative@1.20.0-pre.5, survival@1.20.0-pre.5
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.API.MathTools.SimplexNoiseOctave.Evaluate(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexOctave.cs:line 414
   at Vintagestory.API.MathTools.SimplexNoise.Noise(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexNoise.cs:line 117
   at Vintagestory.GameContent.WeatherPattern.RegenCloudTileCache(Vec3i tileOffset) in VSEssentials\Systems\Weather\Model\Impl\WeatherPattern.cs:line 89
   at Vintagestory.GameContent.WeatherDataReaderBase.ensureCloudTileCacheIsFresh(Vec3i tilePos) in VSEssentials\Systems\Weather\WeatherDataReader.cs:line 275
   at Vintagestory.GameContent.CloudRenderer.UpdateCloudTilesOffThread(Int32 changeSpeed) in VSEssentials\Systems\Weather\CloudRenderer.cs:line 390
   at Vintagestory.GameContent.CloudRenderer.<.ctor>b__31_2() in VSEssentials\Systems\Weather\CloudRenderer.cs:line 166
   ```

**Comment by Craluminum2413** [comment]
To reproduce, type /time speed 50000

**Comment by cheva** [comment]
Game Version: v1.20.0-pre.6 (Unstable)
25-Oct-24 17:01:10: Critical error occurred
Loaded Mods: aged@1.0.4, bigbag@0.1.21-a, buzzwords@1.7.0, caninae@1.0.19, capreolinae@1.1.8, casuariidae@1.0.11, dinornithidae@1.0.1, machairodontinae@1.0.12, pantherinae@1.1.12, rhinocerotidae@1.0.6, sirenia@1.0.8, juicyores@1.0.0, primitivesurvival@3.6.9, game@1.20.0-pre.6, vtr@4.0.4, egocaribautomapmarkers@4.0.0, blocksoverlay@4.0.0-pre.1, carryon@1.8.0-pre.1, danatweaks@3.3.0, extrainfo@1.8.1, localrespawns@1.0.7, nemi@1.1.0, rivers@3.2.0, stepupcontinued@0.0.1, creative@1.20.0-pre.6, vsimgui@1.1.6, survival@1.20.0-pre.6
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.API.MathTools.SimplexNoiseOctave.Evaluate(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexOctave.cs:line 414
   at Vintagestory.API.MathTools.SimplexNoise.Noise(Double x, Double y, Double z) in VintagestoryApi\Math\Noise\SimplexNoise.cs:line 117
   at Vintagestory.GameContent.WeatherPattern.RegenCloudTileCache(Vec3i tileOffset)
   at Vintagestory.GameContent.WeatherDataReaderBase.ensureCloudTileCacheIsFresh(Vec3i tilePos)
   at Vintagestory.GameContent.CloudRenderer.UpdateCloudTilesOffThread(Int32 changeSpeed)
   at Vintagestory.GameContent.CloudRenderer.<.ctor>b__31_2()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)

**Comment by cheva** [comment]
Too many times. With or without mods/ From pre-2 to pre-6.
OS Name	Microsoft Windows 10 Home
Version	10.0.19045 Build 19045
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
System Name	DESKTOP-UHGPKDF
System Manufacturer	LENOVO
System Model	20F5S8FQ00
System Type	x64-based PC
System SKU	LENOVO_MT_20F5_BU_Think_FM_ThinkPad X260
Processor	Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz, 2496 Mhz, 2 Core(s), 4 Logical Processor(s)
BIOS Version/Date	LENOVO R02ET76W (1.49 ), 23-Feb-22
SMBIOS Version	2.8
Embedded Controller Version	1.16
BIOS Mode	UEFI
BaseBoard Manufacturer	LENOVO
BaseBoard Product	20F5S8FQ00
BaseBoard Version	SDK0J40709 WIN
Platform Role	Mobile
Secure Boot State	Off
PCR7 Configuration	Elevation Required to View
Windows Directory	C:\WINDOWS
System Directory	C:\WINDOWS\system32
Boot Device	\Device\HarddiskVolume2
Locale	Russia
Hardware Abstraction Layer	Version = "10.0.19041.3636"
User Name	DESKTOP-UHGPKDF\dmytr
Time Zone	FLE Daylight Time
Installed Physical Memory (RAM)	16.0 GB
Total Physical Memory	15.9 GB
Available Physical Memory	11.0 GB
Total Virtual Memory	21.1 GB
Available Virtual Memory	13.6 GB
Page File Space	5.25 GB
Page File	C:\pagefile.sys
Kernel DMA Protection	Off
Virtualization-based security	Running
Virtualization-based security Required Security Properties	
Virtualization-based security Available Security Properties	Base Virtualization Support, DMA Protection
Virtualization-based security Services Configured	
Virtualization-based security Services Running	
Device Encryption Support	Elevation Required to View
A hypervisor has been detected. Features required for Hyper-V will not be displayed.

**Comment by maltiez2** [comment]
Will be fixed in next update
